frontend/src/screens/auth/SignUp/SignUp.tsx

```diff
@@ -1,29 +1,82 @@
 import styles from "./SignUp.module.css";
-import { Link } from "react-router-dom";
+import { Link, useNavigate } from "react-router-dom";
+import { useState } from "react";
 
 const SignUp = () => {
+  const [email, setEmail] = useState("");
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+  const navigate = useNavigate();
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    try {
+      const response = await fetch(
+        `${import.meta.env.VITE_BACKEND_URL}/auth/signup`,
+        {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({ email, username, password }),
+        }
+      );
+
+      console.log("Response: ", response);
```

Comment:
I would not commit the console statements on the client side to the codebase, imo we should try to mimic a real production environment. Feel free to due something like a toast or alert if there are issues signing up

Reply:
So when initially signing a token, you can set an expiration time on it that would invalidate the token after the specified time. For a token refresh, you can renew the timer on it to allow the user to keep their session without having to log in again if they are still actively using the website. It's kinda like how facebook doesn't require you to login most times when you visit it. That's because the token expiration is pretty long. At least, I assume that is what is done at Facebook. I am yet to implement the refresh token feature though